draft-ietf-mls-protocol.md

```diff
@@ -2230,65 +2230,49 @@ that the path from P to the root may contain some blank nodes that are not
 fixed by P's Parent Hash. However, for each node that has an HPKE key, this key
 is fixed by P's Parent Hash.
 
-Finally, `original_sibling_tree_hash` is the original tree hash of S. The
-original tree hash corresponds to the tree hash of S the last time P was
-updated. It can be computed as the tree hash of S in the ratchet tree modified
-the following way:
+Finally, `original_sibling_tree_hash` is the tree hash of S in the ratchet tree
+modified as follows:
 
-* reset the leaves in P.unmerged_leaves to blanks
-* remove P.unmerged_leaves from all unmerged_leaves lists
-* truncate the ratchet tree as described in {{remove}}
+* Extend the subtree of S until it is full (see {{adding-and-removing-leaves}}).
+* For each leaf L in `P.unmerged_leaves`, blank L and remove it from the
+`unmerged_leaves` sets of all parent nodes.
+
+Observe that `original_sibling_tree_hash` does not change between updates of P.
+This property is crucial for the correctness of the protocol.
 
 For example, in the following tree:
 
 ~~~ ascii-art
-              W [D, H]
+              W [F]
         ______|_____
        /             \
-      U [D]           Y [F, H]
+      U               Y [F]
     __|__           __|__
    /     \         /     \
-  T       _       X [F]   _
- / \     / \     / \     / \
-A   B   C   D   E   F   G   H
+  T       _       _      |
```

Comment:
```suggestion
  T       _       X[F]   |
```

I think this is what you mean if you want to get the result below?

Reply:
Actually, it's the result below that's wrong. This tree couldn't result from any execution. Reason: after the "omit redundant nodes" PR, X would never be inserted by E's commit if F was blank at the time. I changed the result.